uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/services/leave-popup.service.js

```diff
@@ -0,0 +1,120 @@
+/* eslint-env amd */
+
+define([
+  'common/lodash',
+  'leave-absences/shared/modules/services',
+  'common/services/notification.service'
+], function (_, services) {
+  'use strict';
+
+  services.factory('LeavePopup', LeavePopupService);
+
+  LeavePopupService.$inject = [
+    '$log', '$rootElement', '$rootScope', '$q', '$uibModal', 'notificationService',
+    'shared-settings', 'DateFormat', 'Session', 'LeaveRequest'
+  ];
+
+  function LeavePopupService ($log, $rootElement, $rootScope, $q, $modal, notification, sharedSettings, DateFormat, Session, LeaveRequest) {
+    $log.debug('LeavePopup');
+
+    return {
+      openModal: openModal,
+      openModalByID: openModalByID
+    };
+
+    /**
+     * Checks if the current logged in contact can see the leave request
+     *
+     * @return {Boolean}
+     */
+    function checkPermissionBeforeOpeningPopup (leaveRequest) {
+      var deferred = $q.defer();
+
+      Session.get()
+        .then(function (value) {
+          return leaveRequest.roleOf(value.contactId);
+        })
+        .then(function (role) {
+          deferred.resolve(role !== 'none');
+        });
+
+      return deferred.promise;
+    }
+
+    /**
+     * Gets leave type.
+     * If leaveTypeParam exits then its a new request, else if request
+     * object exists then its edit request call
+     *
+     * @param {String} leaveTypeParam
+     * @param {Object} request leave request for edit calls
+     *
+     * @return {String} leave type
+     */
+    function getLeaveType (leaveTypeParam, request) {
+      // reset for edit calls
+      if (request) {
+        return request.request_type;
+      } else if (leaveTypeParam) {
+        return leaveTypeParam;
+      }
+    }
+
+    /**
+     * Open leave request popup for the given leave request
+     *
+     * @param {Object} leaveRequest
+     * @param {String} leaveType
+     * @param {String} selectedContactId
+     * @param {Boolean} isSelfRecord
```

Comment:
these two params *definitely* need some explanation. Would you be able to understand what they are for at a quick glance?